parser/parse.go

```diff
@@ -145,12 +163,8 @@ PACKETLOOP:
 
 		// This means we did not attempt to parse a DNS payload
 		if msg == nil {
-			// Let's check if we had any errors decoding any of the packet layers
```

Comment:
This should not be removed. If `msg == nil` it does not imply that there was an error parsing the packet. It's possible that we simply parsed a TCP packet. This should be reverted when switching back to using `packetSource`.

Reply:
I think this check may be unnecessarily:

When a TCP layer is decoded, the code will either goto `PACKETLOOP` in the for loop or set `msg = new (dns.Msg)`. The only time that `msg == nil` is when neither a TCP nor UDP packet is found.
As for the ErrorLayer check, `DecodingLayerParser.DecodeLayers()` will halt and return the any error that a Layer throws during decoding, so this is still covered.
https://github.com/google/gopacket/blob/master/packet.go#L529